### PR TITLE
feat: add support for `egctl x status`

### LINKED ISSUE #2463: Surface resource status summary in egctl
*Description*:
>Describe the desired behavior, what scenario it enables and how it
would be used.

As a user, Id like to use egtcl to give me a summary of the status of all resources so I can quickly find out 
if resources have be translated and programmed correctly into the data plane

Suggestions for UX
* egctl x status
OR
* egctl x analyze

Relates to https://github.com/envoyproxy/gateway/issues/2028

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

feature: add `egctl x status` subcommand

**What this PR does / why we need it**:

some examples:

image

image

**Which issue(s) this PR fixes**:

Fixes #2463

### CODE REVIEW

File: internal/cmd/egctl/status.go [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+func runStatus(ctx context.Context, cli client.Client, table *tabwriter.Writer, resourceType, namespace string, quiet, verbose, allNamespaces bool) error {
+	var resourcesList client.ObjectList
+
+	if allNamespaces {
+		namespace = ""
+	}
+
+	switch strings.ToLower(resourceType) {
+	case "gc", "gcs", "gatewayclass", "gatewayclasses":
```
Comment (arkodg) [inline internal/cmd/egctl/status.go:93]: lets only use singular terms similar to `kubectl get`

Comment (arkodg) [inline internal/cmd/egctl/status.go:93]:
```suggestion
	case "gc", "gatewayclass":
```

Comment (arkodg) [reply internal/cmd/egctl/status.go:93]: lets do this for all APIs

File: internal/cmd/egctl/status.go [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+}
+
+func runStatus(ctx context.Context, cli client.Client, table *tabwriter.Writer, resourceType, namespace string, quiet, verbose, allNamespaces bool) error {
+	var resourcesList client.ObjectList
+
+	if allNamespaces {
+		namespace = ""
+	}
+
+	switch strings.ToLower(resourceType) {
```
Comment (arkodg) [inline internal/cmd/egctl/status.go:92]: something for the future, dont want to block this PR, but `all` can be helpful here to fetch statuses of all CRDs at once

**Comment by arkodg** [comment]
this PR looks good @shawnh ! added a minor comment

File: internal/cmd/egctl/status.go (hunk 1 above)
Comment (shawnh2) [reply internal/cmd/egctl/status.go]: done

File: internal/cmd/egctl/status.go (hunk 2 above)
Comment (shawnh2) [reply internal/cmd/egctl/status.go:92]: sure, i will add this in a follow-up PR

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**